design-library/src/components/BccGraphic/BccGraphicPoster.vue

```diff
@@ -1,23 +1,70 @@
+<script lang="ts">
+export const ratioClasses = {
+  ultraWide: "42.85%", // 21:9
+  wide: "56.25%", // 16:9
+  landscape: "75%", // 4:3
+  square: "100%", // 1:1
+  portrait: "133.33%", // 3:4
+};
+
+export const roundingClasses = {
+  none: "",
+  sm: "rounded-sm",
+  base: "rounded",
+  md: "rounded-md",
+  xl: "rounded-xl",
+};
+</script>
+
 <script setup lang="ts">
+import { ref } from "vue";
+
 type Props = {
   bannerSrc?: string;
   logoSrc?: string;
-  rounding?: "md" | "xl";
+  rounding?: keyof typeof roundingClasses;
+  ratio?: keyof typeof ratioClasses;
 };
 
 withDefaults(defineProps<Props>(), {
   rounding: "xl",
+  ratio: "wide",
 });
+
+const loadedBanner = ref(false);
+const loadedLogo = ref(false);
 </script>
 
 <template>
-  <div class="relative h-full">
-    <div
-      class="bcc-graphic-poster"
-      :class="{ 'bcc-graphic-poster-rounding-md': rounding === 'md' }"
-      :style="[bannerSrc ? `background-image: url(${bannerSrc});` : '']"
-    >
-      <img class="h-3/4" v-if="logoSrc" :src="logoSrc" />
-    </div>
+  <div class="bcc-graphic-poster" :style="`padding-bottom: ${ratioClasses[ratio]}`">
+    <template v-if="bannerSrc">
+      <img
+        v-show="loadedBanner"
+        @load="loadedBanner = true"
+        :src="bannerSrc"
+        defer
+        class="absolute inset-0 h-full w-full object-cover object-center"
```

Comment:
This should probably become a class as well.

Reply:
Done thanks.